components/automate-ui/src/app/pages/project/confirm-apply-stop-modal/confirm-apply-stop-modal.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, Input, OnChanges, Output, EventEmitter } from '@angular/core';
+import { ApplyRulesStatus } from 'app/entities/projects/project.reducer';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-confirm-apply-stop-modal',
+  templateUrl: './confirm-apply-stop-modal.component.html',
+  styleUrls: [ './confirm-apply-stop-modal.component.scss' ]
+})
+export class ConfirmApplyStopModalComponent implements OnChanges {
+
+  @Input() applyRulesStatus: ApplyRulesStatus;
+
+  @Output() confirm: EventEmitter<void> = new EventEmitter();
+
+  @Output() cancel: EventEmitter<void> = new EventEmitter();
+
+  progressValue: number;
+
+  progressPrefixText: string;
+
+  progressSuffixText: string;
+
+  ngOnChanges() {
+    const { percentageComplete, estimatedTimeComplete } = this.applyRulesStatus;
+    const now = moment();
+    const etc = moment(estimatedTimeComplete);
+    const dur = moment.duration(etc.diff(now));
+    const durHours = Math.floor(dur.get('hours')).toString().padStart(2, '0');
```

Comment:
Don't see why you need `Math.floor` here...? (And why for hours but not for minutes or seconds.)

Reply:
@msorens ah oops, I actually forgot to change this back to using `dur.as('hours')` for hours. `.get` doesn't return what I need for hours.

So the countdown string is setup as `hours:mins:secs` remaining. For mins and secs remaining, we use `.get` method:
```js
dur.get('seconds')  // <= returns num between 0-59
dur.get('minutes')  // <= returns num between 0-59
dur.get('hours')    // <= returns num between 0-23

// 16:51:17 remaining
// 16 hours, 51 mins, 17 secs remaining
```
^ using `.get` method for hours is fine as well if duration is less than 24 hours but if we want the hours to display beyond 24 hours then we want to use `.as` method for hours and round off the remainder (to be displayed by mins and secs):

```js
dur.get('seconds')  // <= returns num between 0-59
dur.get('minutes')  // <= returns num between 0-59
dur.as('hours')     // <= returns length of duration in hours as float

dur.as('hours') // => 36.854893...

// 36:51:17 remaining
// 36 hours, 51 mins, 17 secs remaining
```
![Screen Shot 2019-07-08 at 3 49 09 PM](https://user-images.githubusercontent.com/479121/60844921-099e9700-a198-11e9-91dc-6b05dbe2c419.png)

Perhaps this is all moot since time remaining for updating projects supposedly should never go above 12 hours 🤔 

Either way, I should add unit tests for this as noted in your other comment. Will add some shortly 👍 

As for the methods used, I have no preference for
`.get('hours') / .as('hours')` vs `.hours() / .asHours()`.
I'll use what y'all prefer 👍

Reply:
Thanks for the detailed note!